src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -178,6 +179,9 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
             {
                 try
                 {
+                    // FeatureManagement assemblies may not be loaded on provider startup, so version information is gathered upon first refresh for telemetry
+                    EnsureFeatureManagementVersionInspected();
```

Comment:
... or `InspectFeatureManagementVersion`? Things can happen. We don't really "ensure" anything.

Reply:
"EnsureInitialized" does ensure initialization, so it's a bit different, but I see your point. In similar cases like ours, people sometimes name the API like DoSomething**Once**. This is a private API, so I'm okay with it as is. It's more about code readability for someone new in the future :)